In [1]:
import pandas as pd

In [2]:
f_iso_all = "iso_all_v1.1.csv"
df_iso_all = pd.read_csv(f_iso_all)
df_iso_all

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0                  CHS.15.1   OR4F29  rna-NM_001005221.2  ENST00000426406.4   
1                  CHS.39.7   SAMD11  rna-NM_001385641.1  ENST00000616016.5   
2                  CHS.40.3    NOC2L     rna-NM_015658.4  ENST00000327044.7   
3                 CHS.41.11  PLEKHN1     rna-NM_032129.3  ENST00000379410.8   
4                  CHS.42.6   KLHL17     rna-NM_198317.3  ENST00000338591.8   
...                     ...      ...                 ...                ...   
194775  hypothetical.2081.1        -  rna-NM_001005182.2  ENST00000642104.1   
194776  hypothetical.2082.1        -  rna-NM_001348233.2  ENST00000540461.2   
194777  hypothetical.2083.1        -  rna-NM_001005166.5  ENST00000610445.2   
194778  hypothetical.2084.1        -  rna-NM_001005468.2  ENST00000641451.2   
194779  hypothetical.2085.1        -  rna-NM_001369203.1  ENST00000494082.3   

        pLDDT introns in mouse  
0        88.2       no_introns  
1        46.4            False  
2        71.1            False  
3        49.4            False  
4        86.3             True  
...       ...              ...  
194775   88.9             True  
194776   83.2            False  
194777   83.5            False  
194778   89.3             True  
194779   75.6            False  

[194780 rows x 6 columns]

## Steparate

In [3]:
# df_iso_all[~df_iso_all["CHESS ID"].str.contains("hypothetical")]

In [4]:
df_iso_all_no_GENE = df_iso_all[df_iso_all["GENE"] == "-"]
df_iso_all_no_GENE

CHESS ID GENE           RefSeq ID         GENCODE ID  \
488              CHS.1462.2    -  rna-NM_001377534.1  ENST00000339355.3   
894              CHS.2950.1    -     rna-NM_003516.3  ENST00000369159.2   
1029          CHS.3243.alt1    -  rna-NM_001394530.1  ENST00000392487.1   
1079             CHS.3339.2    -     rna-NM_182679.3  ENST00000368232.9   
1483             CHS.4805.2    -     rna-NM_033445.3                  -   
...                     ...  ...                 ...                ...   
194775  hypothetical.2081.1    -  rna-NM_001005182.2  ENST00000642104.1   
194776  hypothetical.2082.1    -  rna-NM_001348233.2  ENST00000540461.2   
194777  hypothetical.2083.1    -  rna-NM_001005166.5  ENST00000610445.2   
194778  hypothetical.2084.1    -  rna-NM_001005468.2  ENST00000641451.2   
194779  hypothetical.2085.1    -  rna-NM_001369203.1  ENST00000494082.3   

        pLDDT introns in mouse  
488      70.7            False  
894      89.6       no_introns  
1029     66.4       no_introns  
1079     56.9             True  
1483     89.3       no_introns  
...       ...              ...  
194775   88.9             True  
194776   83.2            False  
194777   83.5            False  
194778   89.3             True  
194779   75.6            False  

[26704 rows x 6 columns]

In [5]:
df_iso_all_yes_GENE = df_iso_all[df_iso_all["GENE"] != "-"]
df_iso_all_yes_GENE

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0                  CHS.15.1   OR4F29  rna-NM_001005221.2  ENST00000426406.4   
1                  CHS.39.7   SAMD11  rna-NM_001385641.1  ENST00000616016.5   
2                  CHS.40.3    NOC2L     rna-NM_015658.4  ENST00000327044.7   
3                 CHS.41.11  PLEKHN1     rna-NM_032129.3  ENST00000379410.8   
4                  CHS.42.6   KLHL17     rna-NM_198317.3  ENST00000338591.8   
...                     ...      ...                 ...                ...   
194446         CHS.59041.27     PNCK                   -  ENST00000434652.5   
194447          CHS.59109.1    SMIM9                   -  ENST00000369529.1   
194448  hypothetical.1794.1    CLIC2                   -                  -   
194449  hypothetical.1794.2    CLIC2                   -                  -   
194450          CHS.59119.5    CLIC2                   -  ENST00000465553.5   

        pLDDT introns in mouse  
0        88.2       no_introns  
1        46.4            False  
2        71.1            False  
3        49.4            False  
4        86.3             True  
...       ...              ...  
194446   91.1             True  
194447   66.9            False  
194448   85.9            False  
194449   43.2            False  
194450   83.4            False  

[168076 rows x 6 columns]

In [6]:
# Check if now rows are missed in the separation
print((df_iso_all_no_GENE.shape[0]+df_iso_all_yes_GENE.shape[0]) == df_iso_all.shape[0])

True


In [7]:
# Write to csv file
f_no = "iso_all_v1.1_no_GENE.csv"
df_iso_all_no_GENE.to_csv(f_no, index=False)

In [8]:
# Write to csv file
f_yes = "iso_all_v1.1_yes_GENE.csv"
df_iso_all_yes_GENE.to_csv(f_yes, index=False)

## Merge results back

In [9]:
f_no_fixed = "iso_all_v1.1_no.GENE_filled.csv"
df_iso_all_no_GENE_fixed = pd.read_csv(f_no_fixed)
print(df_iso_all_no_GENE_fixed.shape[0] == df_iso_all_no_GENE.shape[0])
df_iso_all_no_GENE_fixed

True


CHESS ID Unnamed: 1           RefSeq ID         GENCODE ID  \
0               CHS.1462.2     DYNLT4  rna-NM_001377534.1  ENST00000339355.3   
1               CHS.2950.1        NaN     rna-NM_003516.3  ENST00000369159.2   
2            CHS.3243.alt1        NaN  rna-NM_001394530.1  ENST00000392487.1   
3               CHS.3339.2    GPATCH4     rna-NM_182679.3  ENST00000368232.9   
4               CHS.4805.2        NaN     rna-NM_033445.3                  -   
...                    ...        ...                 ...                ...   
26699  hypothetical.2081.1      OR6C1  rna-NM_001005182.2  ENST00000642104.1   
26700  hypothetical.2082.1      OR6J1  rna-NM_001348233.2  ENST00000540461.2   
26701  hypothetical.2083.1     OR52E5  rna-NM_001005166.5  ENST00000610445.2   
26702  hypothetical.2084.1      OR8B2  rna-NM_001005468.2  ENST00000641451.2   
26703  hypothetical.2085.1     SMIM40  rna-NM_001369203.1  ENST00000494082.3   

       pLDDT introns in mouse  
0       70.7            False  
1       89.6       no_introns  
2       66.4       no_introns  
3       56.9             True  
4       89.3       no_introns  
...      ...              ...  
26699   88.9             True  
26700   83.2            False  
26701   83.5            False  
26702   89.3             True  
26703   75.6            False  

[26704 rows x 6 columns]

In [10]:
df_iso_all_no_GENE_fixed.columns = df_iso_all_no_GENE.columns
df_iso_all_no_GENE_fixed

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0               CHS.1462.2   DYNLT4  rna-NM_001377534.1  ENST00000339355.3   
1               CHS.2950.1      NaN     rna-NM_003516.3  ENST00000369159.2   
2            CHS.3243.alt1      NaN  rna-NM_001394530.1  ENST00000392487.1   
3               CHS.3339.2  GPATCH4     rna-NM_182679.3  ENST00000368232.9   
4               CHS.4805.2      NaN     rna-NM_033445.3                  -   
...                    ...      ...                 ...                ...   
26699  hypothetical.2081.1    OR6C1  rna-NM_001005182.2  ENST00000642104.1   
26700  hypothetical.2082.1    OR6J1  rna-NM_001348233.2  ENST00000540461.2   
26701  hypothetical.2083.1   OR52E5  rna-NM_001005166.5  ENST00000610445.2   
26702  hypothetical.2084.1    OR8B2  rna-NM_001005468.2  ENST00000641451.2   
26703  hypothetical.2085.1   SMIM40  rna-NM_001369203.1  ENST00000494082.3   

       pLDDT introns in mouse  
0       70.7            False  
1       89.6       no_introns  
2       66.4       no_introns  
3       56.9             True  
4       89.3       no_introns  
...      ...              ...  
26699   88.9             True  
26700   83.2            False  
26701   83.5            False  
26702   89.3             True  
26703   75.6            False  

[26704 rows x 6 columns]

In [11]:
# Replace NaN with -
df_iso_all_no_GENE_fixed["GENE"] = df_iso_all_no_GENE_fixed["GENE"].fillna("-")
df_iso_all_no_GENE_fixed

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0               CHS.1462.2   DYNLT4  rna-NM_001377534.1  ENST00000339355.3   
1               CHS.2950.1        -     rna-NM_003516.3  ENST00000369159.2   
2            CHS.3243.alt1        -  rna-NM_001394530.1  ENST00000392487.1   
3               CHS.3339.2  GPATCH4     rna-NM_182679.3  ENST00000368232.9   
4               CHS.4805.2        -     rna-NM_033445.3                  -   
...                    ...      ...                 ...                ...   
26699  hypothetical.2081.1    OR6C1  rna-NM_001005182.2  ENST00000642104.1   
26700  hypothetical.2082.1    OR6J1  rna-NM_001348233.2  ENST00000540461.2   
26701  hypothetical.2083.1   OR52E5  rna-NM_001005166.5  ENST00000610445.2   
26702  hypothetical.2084.1    OR8B2  rna-NM_001005468.2  ENST00000641451.2   
26703  hypothetical.2085.1   SMIM40  rna-NM_001369203.1  ENST00000494082.3   

       pLDDT introns in mouse  
0       70.7            False  
1       89.6       no_introns  
2       66.4       no_introns  
3       56.9             True  
4       89.3       no_introns  
...      ...              ...  
26699   88.9             True  
26700   83.2            False  
26701   83.5            False  
26702   89.3             True  
26703   75.6            False  

[26704 rows x 6 columns]

In [13]:
# Merge the df which originally were not missing GENE and the fixed formally-missing df
df_result = pd.concat([df_iso_all_yes_GENE, df_iso_all_no_GENE_fixed])
print(df_result.shape[0] == df_iso_all.shape[0])
df_result

True


CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0                 CHS.15.1   OR4F29  rna-NM_001005221.2  ENST00000426406.4   
1                 CHS.39.7   SAMD11  rna-NM_001385641.1  ENST00000616016.5   
2                 CHS.40.3    NOC2L     rna-NM_015658.4  ENST00000327044.7   
3                CHS.41.11  PLEKHN1     rna-NM_032129.3  ENST00000379410.8   
4                 CHS.42.6   KLHL17     rna-NM_198317.3  ENST00000338591.8   
...                    ...      ...                 ...                ...   
26699  hypothetical.2081.1    OR6C1  rna-NM_001005182.2  ENST00000642104.1   
26700  hypothetical.2082.1    OR6J1  rna-NM_001348233.2  ENST00000540461.2   
26701  hypothetical.2083.1   OR52E5  rna-NM_001005166.5  ENST00000610445.2   
26702  hypothetical.2084.1    OR8B2  rna-NM_001005468.2  ENST00000641451.2   
26703  hypothetical.2085.1   SMIM40  rna-NM_001369203.1  ENST00000494082.3   

       pLDDT introns in mouse  
0       88.2       no_introns  
1       46.4            False  
2       71.1            False  
3       49.4            False  
4       86.3             True  
...      ...              ...  
26699   88.9             True  
26700   83.2            False  
26701   83.5            False  
26702   89.3             True  
26703   75.6            False  

[194780 rows x 6 columns]

In [14]:
df_result = df_result.reset_index(drop=True)
df_result

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0                  CHS.15.1   OR4F29  rna-NM_001005221.2  ENST00000426406.4   
1                  CHS.39.7   SAMD11  rna-NM_001385641.1  ENST00000616016.5   
2                  CHS.40.3    NOC2L     rna-NM_015658.4  ENST00000327044.7   
3                 CHS.41.11  PLEKHN1     rna-NM_032129.3  ENST00000379410.8   
4                  CHS.42.6   KLHL17     rna-NM_198317.3  ENST00000338591.8   
...                     ...      ...                 ...                ...   
194775  hypothetical.2081.1    OR6C1  rna-NM_001005182.2  ENST00000642104.1   
194776  hypothetical.2082.1    OR6J1  rna-NM_001348233.2  ENST00000540461.2   
194777  hypothetical.2083.1   OR52E5  rna-NM_001005166.5  ENST00000610445.2   
194778  hypothetical.2084.1    OR8B2  rna-NM_001005468.2  ENST00000641451.2   
194779  hypothetical.2085.1   SMIM40  rna-NM_001369203.1  ENST00000494082.3   

        pLDDT introns in mouse  
0        88.2       no_introns  
1        46.4            False  
2        71.1            False  
3        49.4            False  
4        86.3             True  
...       ...              ...  
194775   88.9             True  
194776   83.2            False  
194777   83.5            False  
194778   89.3             True  
194779   75.6            False  

[194780 rows x 6 columns]

## Confirm results
- Use mart_export.txt
- Manually check 5 results: CHS.1462.2, CHS.3339.2 ,CHS.55178.1, CHS.24401.13
- 	, , 

In [15]:
df_iso_all_no_GENE.sample(10)

CHESS ID GENE           RefSeq ID         GENCODE ID  \
188063  hypothetical.1115.1    -                   -                  -   
138802      CHS.41131.alt11    -                   -                  -   
121805         CHS.35451.20    -                   -  ENST00000403121.5   
74339        CHS.18996.alt2    -                   -  ENST00000586716.1   
173354         CHS.56701.10    -     rna-NR_104603.2  ENST00000460253.1   
127419       CHS.37184.alt1    -                   -  ENST00000683708.1   
50353          CHS.10638.14    -     rna-NR_027949.3  ENST00000545948.5   
139060         CHS.41249.48    -                   -  ENST00000442707.1   
112557         CHS.31911.17    -  rna-NM_001363913.2                  -   
127552         CHS.37213.28    -                   -  ENST00000497627.1   

        pLDDT introns in mouse  
188063   59.0            False  
138802   59.5             True  
121805   80.8             True  
74339    44.5             True  
173354   80.7            False  
127419   76.3             True  
50353    72.7             True  
139060   60.5             True  
112557   49.9            False  
127552   63.6             True

In [16]:
df_result[df_result["CHESS ID"] == "CHS.1462.2"]
# Confirmed: 

CHESS ID    GENE           RefSeq ID         GENCODE ID  pLDDT  \
168076  CHS.1462.2  DYNLT4  rna-NM_001377534.1  ENST00000339355.3   70.7   

       introns in mouse  
168076            False

In [17]:
df_result[df_result["CHESS ID"] == "CHS.3339.2"]
# Confirmed: 

CHESS ID     GENE        RefSeq ID         GENCODE ID  pLDDT  \
168079  CHS.3339.2  GPATCH4  rna-NM_182679.3  ENST00000368232.9   56.9   

       introns in mouse  
168079             True

In [18]:
df_result[df_result["CHESS ID"] == "CHS.55178.1"]
# Confirmed: 

CHESS ID   GENE        RefSeq ID         GENCODE ID  pLDDT  \
189901  CHS.55178.1  IFNA4  rna-NM_021068.3  ENST00000421715.2   83.2   

       introns in mouse  
189901       no_introns

In [19]:
df_result[df_result["CHESS ID"] == "CHS.24401.13"]
# Confirmed: 

CHESS ID  GENE RefSeq ID         GENCODE ID  pLDDT  \
178675  CHS.24401.13  DSG1         -  ENST00000462981.2   42.2   

       introns in mouse  
178675             True

In [20]:
df_result[df_result["CHESS ID"] == "CHS.345.29"]
# Confirmed: 

CHESS ID   GENE RefSeq ID         GENCODE ID  pLDDT introns in mouse
168312  CHS.345.29  UBE4B         -  ENST00000487244.5   91.4             True

DYNLT4
GPATCH4
IFNA4
DSG1
UBE4B
- grep GENCODE_ID mart_export.txt (in ~/clustering)

### Result confirmed? YES

## Save results in a csv

In [21]:
df_result

CHESS ID     GENE           RefSeq ID         GENCODE ID  \
0                  CHS.15.1   OR4F29  rna-NM_001005221.2  ENST00000426406.4   
1                  CHS.39.7   SAMD11  rna-NM_001385641.1  ENST00000616016.5   
2                  CHS.40.3    NOC2L     rna-NM_015658.4  ENST00000327044.7   
3                 CHS.41.11  PLEKHN1     rna-NM_032129.3  ENST00000379410.8   
4                  CHS.42.6   KLHL17     rna-NM_198317.3  ENST00000338591.8   
...                     ...      ...                 ...                ...   
194775  hypothetical.2081.1    OR6C1  rna-NM_001005182.2  ENST00000642104.1   
194776  hypothetical.2082.1    OR6J1  rna-NM_001348233.2  ENST00000540461.2   
194777  hypothetical.2083.1   OR52E5  rna-NM_001005166.5  ENST00000610445.2   
194778  hypothetical.2084.1    OR8B2  rna-NM_001005468.2  ENST00000641451.2   
194779  hypothetical.2085.1   SMIM40  rna-NM_001369203.1  ENST00000494082.3   

        pLDDT introns in mouse  
0        88.2       no_introns  
1        46.4            False  
2        71.1            False  
3        49.4            False  
4        86.3             True  
...       ...              ...  
194775   88.9             True  
194776   83.2            False  
194777   83.5            False  
194778   89.3             True  
194779   75.6            False  

[194780 rows x 6 columns]

In [23]:
f_result = "iso_all_v1.1_GENE.filled.csv"
df_result.to_csv(f_result, index=False) # None also works

### Confirm that number of transcripts missing GENE is reduced from the original file to the filled file

In [27]:
df_iso_all[df_iso_all["GENE"] == "-"]

CHESS ID GENE           RefSeq ID         GENCODE ID  \
488              CHS.1462.2    -  rna-NM_001377534.1  ENST00000339355.3   
894              CHS.2950.1    -     rna-NM_003516.3  ENST00000369159.2   
1029          CHS.3243.alt1    -  rna-NM_001394530.1  ENST00000392487.1   
1079             CHS.3339.2    -     rna-NM_182679.3  ENST00000368232.9   
1483             CHS.4805.2    -     rna-NM_033445.3                  -   
...                     ...  ...                 ...                ...   
194775  hypothetical.2081.1    -  rna-NM_001005182.2  ENST00000642104.1   
194776  hypothetical.2082.1    -  rna-NM_001348233.2  ENST00000540461.2   
194777  hypothetical.2083.1    -  rna-NM_001005166.5  ENST00000610445.2   
194778  hypothetical.2084.1    -  rna-NM_001005468.2  ENST00000641451.2   
194779  hypothetical.2085.1    -  rna-NM_001369203.1  ENST00000494082.3   

        pLDDT introns in mouse  
488      70.7            False  
894      89.6       no_introns  
1029     66.4       no_introns  
1079     56.9             True  
1483     89.3       no_introns  
...       ...              ...  
194775   88.9             True  
194776   83.2            False  
194777   83.5            False  
194778   89.3             True  
194779   75.6            False  

[26704 rows x 6 columns]

In [28]:
df_result[df_result["GENE"] == "-"]

CHESS ID GENE           RefSeq ID         GENCODE ID  \
168077           CHS.2950.1    -     rna-NM_003516.3  ENST00000369159.2   
168078        CHS.3243.alt1    -  rna-NM_001394530.1  ENST00000392487.1   
168080           CHS.4805.2    -     rna-NM_033445.3                  -   
168081           CHS.4806.2    -     rna-NM_175055.3                  -   
168087  hypothetical.2088.1    -  rna-NM_001365552.1                  -   
...                     ...  ...                 ...                ...   
194756  hypothetical.2062.1    -       id-TRBVAOR9-2                  -   
194764  hypothetical.2070.1    -           id-TRGV10                  -   
194765  hypothetical.2071.1    -           id-TRGV11                  -   
194766  hypothetical.2072.1    -            id-TRGVA                  -   
194767  hypothetical.2073.1    -            id-TRGVB                  -   

        pLDDT introns in mouse  
168077   89.6       no_introns  
168078   66.4       no_introns  
168080   89.3       no_introns  
168081   84.4       no_introns  
168087   57.4            False  
...       ...              ...  
194756   30.9       no_introns  
194764   88.0            False  
194765   69.3            False  
194766   58.2            False  
194767   39.2            False  

[16233 rows x 6 columns]

In [29]:
n_filled = df_iso_all[df_iso_all["GENE"] == "-"].shape[0] - df_result[df_result["GENE"] == "-"].shape[0]
print(str(n_filled) + " transcripts had their GENE filled using GENCODE&mart_export.txt")

10471 transcripts had their GENE filled using GENCODE&mart_export.txt


Yes, it has reduced (by 10471)

### Check whether number of types of GENES increased

In [30]:
print("In original iso_all, there are " + str(len(df_iso_all["GENE"].unique())) + " types of genes")

In original iso_all, there are 16124 types of genes


In [31]:
print("In GENE filled iso_all, there are " + str(len(df_result["GENE"].unique())) + " types of genes")

In GENE filled iso_all, there are 19231 types of genes


Yes, there are about 3107 more types of genes after filling missing gaps using GENCODE ID and mart_export.txt